project_management_system/src/app/auth/login/login.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import {
+  FormBuilder,
+  FormGroup,
+  FormControl,
+  Validators,
+} from '@angular/forms';
+import { first } from 'rxjs/operators';
+import { AccountService } from '../../services/account.service';
+import { AlertService } from '../../services/alert.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent implements OnInit {
+  form!: FormGroup;
+  loading = false;
+  submitted = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private route: ActivatedRoute,
+    private router: Router,
+    private accountService: AccountService,
+    private alertService: AlertService
+  ) {}
+
+  ngOnInit() {
+    this.form = new FormGroup({
+      login: new FormControl<string>('', [Validators.required]),
+      password: new FormControl<string>('', [Validators.required]),
+    });
+  }
+
+  get f() {
```

Comment:
f -  is not informative. It better to use the names of methods or property that clarify what it does. That may help you or your colleagues to read and understand your code